--- /usr/local/captiveportal/index.php.original  2022-01-31 16:53:42.000000000 -0300
+++ /usr/local/captiveportal/index.php   2022-09-01 22:01:09.736907000 -0300
@@ -38,6 +38,7 @@
 global $cpzone, $cpzoneid;

 $cpzone = strtolower($_REQUEST['zone']);
+if(empty($cpzone)){ $cpzone = array_keys($config["captiveportal"])[0]; }
 $cpcfg = $config['captiveportal'][$cpzone];

 /* NOTE: IE 8/9 is buggy and that is why this is needed */
